app/_src/mesh/features/provenance-verification-binaries.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Verify Build Provenance for Kong Mesh Binaries
+badge: enterprise
+---
+
+Starting with 2.7.4, {{site.mesh_product_name}} produces build provenance for binary artifacts, which can be verified using `cosign` / `slsa-verifier` with attestations published to a Docker Hub repository.
+
+This guide provides steps to verify build provenance for signed {{site.mesh_product_name}} binary artifacts in two different ways:
+
+* A minimal example, used to verify an binary artifacts without leveraging any annotations
+* A complete example, leveraging optional annotations for increased trust
+
+For the minimal example, you only need a compressed binary file and provenance file.
+
+For the complete example, you need the same details as the minimal example, as well as any of the optional annotations you wish to verify:
+
+| Shorthand | Description | Example Value |
+|---|---|---|
+| `<repo>` | GitHub repository | `kong-mesh` |
+| `<workflow name>` | GitHub workflow name | `build-test-distribute` |
+| `<workflow trigger>` | Github workflow trigger name | `push` |
+| `<version>` | Artifact version to download | `2.7.4` |
+| `<binary-files>` | Compressed binary files for the specified version | `kong-mesh-2.7.4-*-*.tar.gz` |
+| `<provenance-file>` | Binary provenance file | `kong-mesh.intoto.jsonl` |
+
+Because Kong uses GitHub Actions to build and release, Kong also uses GitHub's OIDC identity to generate build provenance for binary artifacts, which is why many of these details are GitHub-related.
+
+## Examples
+
+### Prerequisites
+
+For both examples, you need to:
+
+1. Ensure `slsa-verifier` is installed.
+
+2. [Download security assets](https://cloudsmith.io/~kong/repos/kong-mesh-binaries-release/packages/?q=name%3Asecurity-assets*+version%3A{{page.kong_latest.version}}) for the required version of {{site.mesh_product_name}} binaries
```

Comment:
Why link to cloudsmith and not directly to the thing to download: https://packages.konghq.com/public/kong-mesh-binaries-release/raw/names/security-assets/versions/2.7.4/security-assets.tar.gz ?

Reply:
Seems like the in-toto is not included but that's fine.

Reply:
I don't see the in-toto in the security-assets is this normal?

Reply:
No, the release tags should have the `intoto.jsonl` within the `security-assets`

Reply:
Let me look into this!!

Reply:
The tag `2.7.4` was using the slsa provenance generator [v1.10.0](https://github.com/Kong/kong-mesh/blob/2.7.4/.github/workflows/kuma-_provenance.yaml#L33C11-L33C101) instead of [v2.0.0](https://github.com/kumahq/kuma/blob/master/.github/workflows/_provenance.yaml#L33). 

An [improvement / breaking change](https://github.com/slsa-framework/slsa-github-generator/blob/v2.0.0/CHANGELOG.md#v200) to use `v4` to allow downloading the artifacts during the workflow run for later jobs instead of waiting until the end of workflow run.

The [step](https://github.com/Kong/kong-mesh/actions/runs/9592834068/job/26455512866#step:4:17) `download-artifact@v4` used in the distributions to upload to cloudsmith  failed to find / filter `kong-mesh.intoto.jsonl` from workflow assets since it was uploaded using an older API of `actions/upload-artifacts` with the issue mentioned above

This should be fixed in [release-2.8](https://github.com/kumahq/kuma/blob/release-2.8/.github/workflows/_provenance.yaml) as it uses the version to make artifacts available to be accessible later in the workflow. NO change is needed.

Reply:
A backport of this might be needed for later 2.7.x minor upgrades. Hence I am moving the doc content for for  Image Provenance and Binary Provenance to be available from: '>=2.8.0'

Reply:
This is fixed and provenance works from 2.8.0